homeassistant/components/maxcul.py

```diff
@@ -0,0 +1,111 @@
+"""
+Support for MAX! devices via a CUL stick.
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/maxcul/
+"""
+import logging
+
+import voluptuous as vol
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+REQUIREMENTS = ['pymaxcul==0.1.9']
+
+DOMAIN = 'maxcul'
+
+CONF_DEVICE_PATH = 'device_path'
+CONF_DEVICE_BAUD_RATE = 'device_baud_rate'
+CONF_DEVICE_ID = 'device_id'
+
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_DEVICE_PATH): cv.string,
+        vol.Required(CONF_DEVICE_BAUD_RATE): cv.positive_int,
+        vol.Optional(CONF_DEVICE_ID): cv.positive_int,
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+DATA_MAXCUL = 'maxcul'
+
+EVENT_THERMOSTAT_UPDATE = DOMAIN + '.thermostat_update'
```

Comment:
Don't use events to communicate updates between component and platforms. use dispatcher